src/fancy-transformers/defragmenter.ts

```diff
@@ -0,0 +1,87 @@
+import logging = require('../logging/logging');
+import arraybuffers = require('../arraybuffers/arraybuffers');
+import Fragment = require('./fragment');
+
+var log :logging.Log = new logging.Log('fancy-transformers');
+
+class Defragmenter {
+  private tracker_ : {[index: string]: ArrayBuffer[]}={};
+  private counter_ : {[index: string]: number}={};
+  private complete_: string[]=[];
+
+  public constructor() {}
+
+  public addFragment(fragment:Fragment) {
+    var hexid=arraybuffers.arrayBufferToHexString(fragment.id);
+    if(hexid in this.tracker_) { // A fragment for an existing packet
```

Comment:
Vertical space is cheap, visual bandwidth is expensive.  Please put the comments on their own lines.